llvm/lib/MCCAS/MCCASObjectV1.cpp

```diff
@@ -3155,25 +3165,151 @@ static AbbrevEntryReader getAbbrevEntryReader(ArrayRef<StringRef> AbbrevEntries,
   return AbbrevEntryReader(AbbrevData);
 }
 
-Error DIEVisitor::visitDIERef(BinaryStreamReader &DataReader,
-                              unsigned AbbrevIdx, StringRef DIEData,
-                              ArrayRef<DIEDataRef> &DIEChildrenStack) {
+static std::optional<uint8_t> getNonULEBFormSize(dwarf::Form Form,
+                                                 dwarf::FormParams FP) {
+  switch (Form) {
+  case dwarf::DW_FORM_addr:
+  case dwarf::DW_FORM_ref_addr: {
+    auto FormSize =
+        (Form == dwarf::DW_FORM_addr) ? FP.AddrSize : FP.getRefAddrByteSize();
+    return FormSize;
+  }
+  case dwarf::DW_FORM_exprloc:
+  case dwarf::DW_FORM_block:
+  case dwarf::DW_FORM_block1:
+  case dwarf::DW_FORM_block2:
+  case dwarf::DW_FORM_block4:
+  case dwarf::DW_FORM_sdata:
+  case dwarf::DW_FORM_udata:
+  case dwarf::DW_FORM_ref_udata:
+  case dwarf::DW_FORM_ref4_cas:
+  case dwarf::DW_FORM_strp_cas:
+  case dwarf::DW_FORM_rnglistx:
+  case dwarf::DW_FORM_loclistx:
+  case dwarf::DW_FORM_GNU_addr_index:
+  case dwarf::DW_FORM_GNU_str_index:
+  case dwarf::DW_FORM_addrx:
+  case dwarf::DW_FORM_strx:
+  case dwarf::DW_FORM_LLVM_addrx_offset:
+  case dwarf::DW_FORM_string:
+  case dwarf::DW_FORM_indirect: {
+    return std::nullopt;
+  }
+
+  case dwarf::DW_FORM_implicit_const:
+  case dwarf::DW_FORM_flag_present: {
+    return 0;
+  }
+  case dwarf::DW_FORM_data1:
+  case dwarf::DW_FORM_ref1:
+  case dwarf::DW_FORM_flag:
+  case dwarf::DW_FORM_strx1:
+  case dwarf::DW_FORM_addrx1: {
+    return 1;
+  }
+  case dwarf::DW_FORM_data2:
+  case dwarf::DW_FORM_ref2:
+  case dwarf::DW_FORM_strx2:
+  case dwarf::DW_FORM_addrx2: {
+    return 2;
+  }
+  case dwarf::DW_FORM_strx3: {
+    return 3;
+  }
+  case dwarf::DW_FORM_data4:
+  case dwarf::DW_FORM_ref4:
+  case dwarf::DW_FORM_ref_sup4:
+  case dwarf::DW_FORM_strx4:
+  case dwarf::DW_FORM_addrx4: {
+    return 4;
+  }
+  case dwarf::DW_FORM_ref_sig8:
+  case dwarf::DW_FORM_data8:
+  case dwarf::DW_FORM_ref8:
+  case dwarf::DW_FORM_ref_sup8: {
+    return 8;
+  }
+  case dwarf::DW_FORM_data16: {
+    return 16;
+  }
+
+  case dwarf::DW_FORM_strp:
+  case dwarf::DW_FORM_sec_offset:
+  case dwarf::DW_FORM_GNU_ref_alt:
+  case dwarf::DW_FORM_GNU_strp_alt:
+  case dwarf::DW_FORM_line_strp:
+  case dwarf::DW_FORM_strp_sup: {
+    auto FormSize = FP.getDwarfOffsetByteSize();
+    return FormSize;
+  }
+  case dwarf::DW_FORM_addrx3:
+  case dwarf::DW_FORM_lo_user: {
+    llvm_unreachable("usupported form");
+    break;
+  }
+  }
+}
+
+Error DIEVisitor::materializeAbbrevDIE(unsigned AbbrevIdx) {
+  constexpr auto AddrSize = 8;
```

Comment:
Where does this come from? Should it be hardcoded?

Reply:
For all of our purposes the AddressSize is always 8 bytes on MachO, if you see `DIEVisitor::visitDIEAttrs`, the same is true there

Reply:
In fact, if you see `getAndSetDebugAbbrevOffsetAndSkip`, we have a check to make sure that the address size is 8

```    
if (AddressSize != 8)
      return createStringError(
          inconvertibleErrorCode(),
          "Address size is not 8 bytes, unsupported architecture for MCCAS!");
```